WoofWare.PawPrint/UnaryMetadataIlOp.fs

```diff
@@ -303,19 +303,38 @@ module internal UnaryMetadataIlOp =
         | Box -> failwith "TODO: Box unimplemented"
         | Ldelema -> failwith "TODO: Ldelema unimplemented"
         | Isinst ->
-            let targetType =
+            let actualObj, state = IlMachineState.popEvalStack thread state
+
+            let targetType : TypeDefn =
                 match metadataToken with
-                | MetadataToken.TypeDefinition td -> state.ActiveAssembly(thread).TypeDefs.[td]
+                | MetadataToken.TypeDefinition td ->
+                    TypeDefn.FromDefinition (
+                        ComparableTypeDefinitionHandle.Make td,
+                        state.ActiveAssembly(thread).Name.FullName,
+                        failwith "TODO"
+                    )
+                | MetadataToken.TypeSpecification handle -> state.ActiveAssembly(thread).TypeSpecs.[handle].Signature
                 | m -> failwith $"unexpected metadata token {m} in IsInst"
 
-            let actualObj, state = IlMachineState.popEvalStack thread state
-
             let returnObj =
                 match actualObj with
                 | EvalStackValue.ManagedPointer ManagedPointerSource.Null ->
                     // null IsInstance check always succeeds and results in a null reference
                     EvalStackValue.ManagedPointer ManagedPointerSource.Null
-                | v -> failwith $"TODO: %O{v}"
+                | EvalStackValue.ManagedPointer (ManagedPointerSource.LocalVariable _) -> failwith "TODO"
```

Comment:
Waaaaa here is where we have to start implementing the `verifier-assignable-to` relation